packages/gotrue/pubspec.yaml

```
@@ -1,26 +1,26 @@
 name: gotrue
 description: A dart client library for the GoTrue API.
 version: 2.16.0
-homepage: 'https://supabase.com'
-repository: 'https://github.com/supabase/supabase-flutter/tree/main/packages/gotrue'
-documentation: 'https://supabase.com/docs/reference/dart/auth-signup'
+homepage: "https://supabase.com"
+repository: "https://github.com/supabase/supabase-flutter/tree/main/packages/gotrue"
+documentation: "https://supabase.com/docs/reference/dart/auth-signup"
 
 environment:
-  sdk: '>=3.3.0 <4.0.0'
+  sdk: ">=3.3.0 <4.0.0"
 
 dependencies:
   collection: ^1.15.0
   crypto: ^3.0.2
-  http: '>=0.13.0 <2.0.0'
+  http: ">=0.13.0 <2.0.0"
   jwt_decode: ^0.3.1
   retry: ^3.1.0
-  rxdart: '>=0.27.7 <0.29.0'
+  rxdart: ">=0.27.7 <0.29.0"
   meta: ^1.7.0
   logging: ^1.2.0
-  web: '>=0.5.0 <2.0.0'
+  web: ">=0.5.0 <2.0.0"
+  dart_jsonwebtoken: ^2.17.0
```

```suggestion
  dart_jsonwebtoken: ">=2.17.0 <4.0.0"
```
I know these versions conflicts are really tedious, be I think we should support if possible the latest version. This constraint should use 3.0.0 for people with at least Dart 3.4.0 and fallback to 2.17.0 for people below that.

---

I don't think that will be possible, as implementation differs significantly, but let me try.